runhouse/resources/hardware/cluster.py

```diff
@@ -502,6 +503,13 @@ def is_up(self) -> bool:
         """
         return self.on_this_cluster() or self._ping()
 
+    def _is_server_up(self) -> bool:
```

Comment:
how come `is_up` isn't enough here?

Reply:
because is_up checking if self.address is not None. But sometimes self.address might be equal to an ip ( == the cluster is live on the cloud), but the runhouse server is down.